### PR TITLE
Add GRPO algorithm

### PR DESCRIPTION
**Pull request by yangchen73**
# Description

Add GRPO (Group Relative Policy Optimization) for RL training. GRPO uses group-level return comparison instead of a Critic network, reducing memory and supporting both Embodied AI and VLA fine-tuning tasks.

## Key Features
- **Step-wise returns**: Per-step discounted returns $R_t = r_t + \gamma R_{t+1}$ with reverse accumulation
- **Masked group normalization**: Variable-length sequences supported; only alive peers at each step used for mean/std
- **Optional reference policy**: When `kl_coef > 0`, KL regularization for VLA fine-tuning; when `kl_coef = 0`, no ref policy (recommended for CartPole, etc.)
- **Config options**: `group_size`, `truncate_at_first_done`, `reset_every_rollout`

## Type of change

- New feature (non-breaking change which adds functionality)
- Documentation update

## Checklist

- [x] I have run the `black .` command to format the code base.
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Dependencies have been updated, if applicable.

### CODE REVIEW

File: docs/source/overview/rl/algorithm.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # RL Algorithms
 
-This module contains the core implementations of reinforcement learning algorithms, mainly including PPO (Proximal Policy Optimization).
+This module contains the core implementations of reinforcement learning algorithms, including PPO (Proximal Policy Optimization) and GRPO (Group Relative Policy Optimization).
```
Comment (yuecideng) [inline docs/source/overview/rl/algorithm.md:3]: We can use a table to show the implemented algorithm

File: embodichain/agents/rl/models/actor_only.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        log_std = self.log_std.clamp(self.log_std_min, self.log_std_max)
+        std = log_std.exp().expand(mean.shape[0], -1)
+        dist = Normal(mean, std)
+        action = mean if deterministic else dist.sample()
+        log_prob = dist.log_prob(action).sum(dim=-1)
+        value = torch.zeros(obs.shape[0], device=self.device, dtype=obs.dtype)
+        return action, log_prob, value
+
+    @torch.no_grad()
+    def get_value(self, obs: torch.Tensor) -> torch.Tensor:
```
Comment (yuecideng) [inline embodichain/agents/rl/models/actor_only.py:67]: We may remove from `Policy` (only implement for ActorCritic)

**Review by yuecideng — CHANGES_REQUESTED** [review]

**Review by yuecideng — APPROVED** [review]
